### ISSUE TITLE
Implement UserALE.js to obtain MUDROD WebUI Metrics

### ISSUE THREAD

**Issue by lewismc** [mudrod/mudrod#212]
[Apache UserALE.js](http://senssoft.apache.org/userale/) will enable us, with a single line of code, capture all MUDROD users’ behavior on the GUI with enough resolution and granularity for advanced behavioral modeling. Apache UserALE is a free, open-source application for standard user event tracking, but is specially tailored for capturing behavior in web applications build for productivity—data analysis, logistics, system monitoring, and service provisioning.

**Comment by quintinali** [comment]
@lewismc This tool sounds very useful. We also want to improve ranking performance from clickthrough data as described in https://www.cs.cornell.edu/people/tj/publications/joachims_02c.pdf.  In summary, a search engine should be able to record the position of clicked pages.  For example, to a query "sst", the search engine returns a list of related metadata and records the position ( page number and position number) of metadata clicked by a user. If most people click the third metadata on the first page, the ranking algorithm will learn that this metadata should have a higher rank.

How do you think about it? Thanks.

**Comment by lewismc** [comment]
YES :+1: I am going to start work on this with the aim of getting the click events streaming into our ES instance @quintinali . Thank you for providing the link to the paper.
